integrations/nvidia/src/haystack_integrations/components/embedders/nvidia/text_embedder.py

```diff
@@ -70,6 +72,33 @@ def __init__(
         self.backend: Optional[EmbedderBackend] = None
         self._initialized = False
 
+        self.is_hosted = urlparse(self.api_url).netloc in [
+            "integrate.api.nvidia.com",
+            "ai.api.nvidia.com",
+        ]
+        if self.is_hosted and not self.model:
+            # manually set default model
+            self.model = "NV-Embed-QA"
```

Comment:
Looks like this can be moved to `warm_up`, removing the need to store `is_hosted`?